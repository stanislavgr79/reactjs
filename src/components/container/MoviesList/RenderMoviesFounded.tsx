import React, { useState } from 'react';

import MovieCard from '../../atom/MovieCard';
import MovieMenuPopup from '../MovieMenuPopup';
import { IMovie } from '../../../helpers/interface';

import image from '../../../resources/images/menu-icon.png';
import './RenderMoviesFounded.scoped.scss';

interface IProp {
  movies: IMovie[];
}

export default function RenderMoviesFounded({ movies }: IProp): JSX.Element {
  const [showSubMenu, setShowSubMenu] = useState(false);
  const [activeMovieId, setActiveMovieId] = useState(0);

  const updateShowSubMenuWithId = (id: number) => {
    setActiveMovieId(id);
    setShowSubMenu(!showSubMenu);
  };

  const updateShowSubMenu = () => {
    setShowSubMenu(!showSubMenu);
  };

  return (
    <>
      <div className="container-md movies-wraper">
        <div className="row row-cols-3">
          {movies.map((movie: IMovie, index: number) => (
            <div className="col movie-unit" key={`unit-${index}`}>
              <div className="movie_holder">
                <MovieCard movie={movie} />
                {!showSubMenu ? (
                  <img
                    src={image}
                    className="sub_menu_icon first"
                    onClick={() => updateShowSubMenuWithId(Number(movie.id))}
                  />
                ) : activeMovieId == movie.id ? (
                  <MovieMenuPopup movie={movie} closeMenu={updateShowSubMenu} />
                ) : (
                  <img src={image} className="sub_menu_icon second" onClick={updateShowSubMenu} />
                )}
              </div>
            </div>
          ))}
        </div>
      </div>
    </>
  );
}
