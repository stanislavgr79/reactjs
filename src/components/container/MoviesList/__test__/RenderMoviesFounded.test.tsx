import React from 'react';
import { render, screen } from '@testing-library/react';

import RenderMoviesFounded from '../RenderMoviesFounded';
import userEvent from '@testing-library/user-event';

jest.mock('../../../atom/MovieCard', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="MovieCard"></div>;
    },
  };
});

jest.mock('../../MovieMenuPopup', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="MovieMenuPopup"></div>;
    },
  };
});

describe('RenderMoviesFounded', () => {
  const movie1 = {
    id: 12345,
    title: 'film title',
    release_date: '2020-01-02',
    poster_path: '',
    genres: ['crime'],
    overview: 'overview',
    runtime: 66,
    vote_average: 5.5,
    tagline: '0',
  };
  const movie2 = {
    id: 98765,
    title: 'film title',
    release_date: '2020-01-02',
    poster_path: '',
    genres: ['crime'],
    overview: 'overview',
    runtime: 66,
    vote_average: 5.5,
    tagline: '0',
  };
  const movies = [movie1];

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('render', () => {
    const tree = render(<RenderMoviesFounded movies={movies} />);

    expect(tree.baseElement).toMatchSnapshot();
  });

  test('render MovieMenuPopup', () => {
    const tree = render(<RenderMoviesFounded movies={movies} />);

    userEvent.click(screen.getByRole('img'));
    expect(tree.baseElement).toMatchSnapshot();
  });

  test('render MovieMenuPopup2', () => {
    const movies = [movie1, movie2];
    const tree = render(<RenderMoviesFounded movies={movies} />);

    const images = screen.getAllByRole('img');
    userEvent.click(images[1]);
    expect(tree.baseElement).toMatchSnapshot();
    userEvent.click(screen.getByRole('img'));
    expect(tree.baseElement).toMatchSnapshot();
  });
});
