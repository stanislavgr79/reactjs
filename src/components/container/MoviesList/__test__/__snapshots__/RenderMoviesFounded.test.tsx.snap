// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderMoviesFounded render 1`] = `
<body>
  <div>
    <div
      class="container-md movies-wraper"
    >
      <div
        class="row row-cols-3"
      >
        <div
          class="col movie-unit"
        >
          <div
            class="movie_holder"
          >
            <div
              class="MovieCard"
            />
            <img
              class="sub_menu_icon first"
              src="[object Object]"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`RenderMoviesFounded render MovieMenuPopup 1`] = `
<body>
  <div>
    <div
      class="container-md movies-wraper"
    >
      <div
        class="row row-cols-3"
      >
        <div
          class="col movie-unit"
        >
          <div
            class="movie_holder"
          >
            <div
              class="MovieCard"
            />
            <div
              class="MovieMenuPopup"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`RenderMoviesFounded render MovieMenuPopup2 1`] = `
<body>
  <div>
    <div
      class="container-md movies-wraper"
    >
      <div
        class="row row-cols-3"
      >
        <div
          class="col movie-unit"
        >
          <div
            class="movie_holder"
          >
            <div
              class="MovieCard"
            />
            <img
              class="sub_menu_icon second"
              src="[object Object]"
            />
          </div>
        </div>
        <div
          class="col movie-unit"
        >
          <div
            class="movie_holder"
          >
            <div
              class="MovieCard"
            />
            <div
              class="MovieMenuPopup"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`RenderMoviesFounded render MovieMenuPopup2 2`] = `
<body>
  <div>
    <div
      class="container-md movies-wraper"
    >
      <div
        class="row row-cols-3"
      >
        <div
          class="col movie-unit"
        >
          <div
            class="movie_holder"
          >
            <div
              class="MovieCard"
            />
            <img
              class="sub_menu_icon first"
              src="[object Object]"
            />
          </div>
        </div>
        <div
          class="col movie-unit"
        >
          <div
            class="movie_holder"
          >
            <div
              class="MovieCard"
            />
            <img
              class="sub_menu_icon first"
              src="[object Object]"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
